File: packages/google_mlkit_genai_summarization/ios/Classes/GoogleMlKitGenaiSummarizationPlugin.m
#import <Foundation/Foundation.h>
#import "GoogleMlKitGenaiSummarizationPlugin.h"

#define channelName @"google_mlkit_genai_summarization"
#define checkFeatureStatus @"genai#checkFeatureStatus"
#define downloadFeature @"genai#downloadFeature"
#define runInference @"genai#runInference"
#define runInferenceStreaming @"genai#runInferenceStreaming"
#define closeSummarizer @"genai#closeSummarizer"

@implementation GoogleMlKitGenaiSummarizationPlugin {
    NSMutableDictionary *instances;
}

+ (void)registerWithRegistrar:(NSObject<FlutterPluginRegistrar>*)registrar {
    FlutterMethodChannel* channel = [FlutterMethodChannel
                                     methodChannelWithName:channelName
                                     binaryMessenger:[registrar messenger]];
    GoogleMlKitGenaiSummarizationPlugin* instance = [[GoogleMlKitGenaiSummarizationPlugin alloc] init];
    [registrar addMethodCallDelegate:instance channel:channel];
}

- (id)init {
    self = [super init];
    if (self)
        instances = [NSMutableDictionary dictionary];
    return self;
}

- (void)handleMethodCall:(FlutterMethodCall *)call result:(FlutterResult)result {
    if ([call.method isEqualToString:checkFeatureStatus]) {
        // iOS implementation would go here
        // Note: GenAI APIs are currently Android-only
        result([FlutterError errorWithCode:@"UNIMPLEMENTED"
                                   message:@"GenAI APIs are currently only available on Android"
                                   details:nil]);
    } else if ([call.method isEqualToString:downloadFeature]) {
        result([FlutterError errorWithCode:@"UNIMPLEMENTED"
                                   message:@"GenAI APIs are currently only available on Android"
                                   details:nil]);
    } else if ([call.method isEqualToString:runInference]) {
        result([FlutterError errorWithCode:@"UNIMPLEMENTED"
                                   message:@"GenAI APIs are currently only available on Android"
                                   details:nil]);
    } else if ([call.method isEqualToString:runInferenceStreaming]) {
        result([FlutterError errorWithCode:@"UNIMPLEMENTED"
                                   message:@"GenAI APIs are currently only available on Android"
                                   details:nil]);
    } else if ([call.method isEqualToString:closeSummarizer]) {
        NSString *uid = call.arguments[@"id"];
        [instances removeObjectForKey:uid];
        result(NULL);
    } else {
        result(FlutterMethodNotImplemented);
    }
}

@end

File: packages/google_mlkit_genai_summarization/ios/google_mlkit_genai_summarization/Sources/google_mlkit_genai_summarization/GoogleMlKitGenaiSummarizationPlugin.swift
import Flutter

@objc
public class GoogleMlKitGenaiSummarizationPlugin: NSObject, FlutterPlugin {
  private var instances: [String: Any] = [:]

  public static func register(with registrar: FlutterPluginRegistrar) {
    let channel = FlutterMethodChannel(
      name: "google_mlkit_genai_summarization",
      binaryMessenger: registrar.messenger()
    )
    let instance = GoogleMlKitGenaiSummarizationPlugin()
    registrar.addMethodCallDelegate(instance, channel: channel)
  }

  public func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
    let unimplemented = FlutterError(
      code: "UNIMPLEMENTED",
      message: "GenAI APIs are currently only available on Android",
      details: nil
    )
    switch call.method {
    case "genai#checkFeatureStatus", "genai#downloadFeature", "genai#runInference", "genai#runInferenceStreaming":
      result(unimplemented)
    case "genai#closeSummarizer":
      if let args = call.arguments as? [String: Any], let uid = args["id"] as? String {
        instances.removeValue(forKey: uid)
      }
      result(nil)
    default:
      result(FlutterMethodNotImplemented)
    }
  }
}
